pkg/config/configuration.go

```diff
@@ -98,6 +98,7 @@ type ConfigurationSpec struct {
 	TracingSpec         TracingSpec        `json:"tracing,omitempty" yaml:"tracing,omitempty"`
 	MTLSSpec            MTLSSpec           `json:"mtls,omitempty" yaml:"mtls,omitempty"`
 	MetricSpec          MetricSpec         `json:"metric,omitempty" yaml:"metric,omitempty"`
```

Comment:
Should this be made a pointer so we can check if it's nil?

Reply:
In effect this can never be nil as we explicitly set it to true by default.